typescript/sdk/src/deploy/HyperlaneDeployer.ts

```diff
@@ -240,27 +242,32 @@ export abstract class HyperlaneDeployer<
   protected async configureHook<C extends Ownable>(
     chain: ChainName,
     contract: C,
-    targetHook: Address,
+    targetHook: IPostDispatchHook,
     getHook: (contract: C) => Promise<Address>,
     setHook: (contract: C, hook: Address) => Promise<PopulatedTransaction>,
   ): Promise<void> {
     const configuredHook = await getHook(contract);
-    if (!eqAddress(targetHook, configuredHook)) {
-      await this.runIfOwner(chain, contract, async () => {
+    if (!eqAddress(targetHook.address, configuredHook)) {
+      const result = await this.runIfOwner(chain, contract, async () => {
         this.logger(
-          `Set hook on ${chain} to ${targetHook}, currently is ${configuredHook}`,
+          `Set hook on ${chain} to ${targetHook.address}, currently is ${configuredHook}`,
         );
         await this.multiProvider.sendTransaction(
           chain,
-          setHook(contract, targetHook),
+          setHook(contract, targetHook.address),
         );
         const actualHook = await getHook(contract);
-        if (!eqAddress(targetHook, actualHook)) {
+        if (!eqAddress(targetHook.address, actualHook)) {
           throw new Error(
-            `Set hook failed on ${chain}, wanted ${targetHook}, got ${actualHook}`,
+            `Set hook failed on ${chain}, wanted ${targetHook.address}, got ${actualHook}`,
           );
         }
+        return true;
       });
+      // if the signer is not the owner, saving the hook address in the artifacts for later use for sending test messages, etc
+      if (!result) {
+        this.addDeployedContracts(chain, { customHook: targetHook });
+      }
```

Comment:
Not directly related to this addition but I'm curious to get your take on the maintainability of the deployers atm. To you feel like we have a flexible enough foundation to build on or is it difficult to support new scenarios?

Reply:
Yorke and I were discussing this IRL. Yorke thinks we need to not deploy the core config on the core chains because we would need to replicate maintain predeployed core artifacts. This means we need to run through some of the `runIfOwner` function needlessly for core chains. I don't feel strongly but do think the deployer should be resilient to these changes.